### PR TITLE
Dependency updates & add errorprone

### PR DESCRIPTION
**Pull request by kingthorin**
- Add errorprone 2.0.2
- Update ben-manes.versions to 0.39.0
- Update spotless to 5.14.1
- Enable spotless for build files

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
@thc202 seem good?

File: build.gradle.kts [hunk 1]
```diff
@@ -22,6 +24,16 @@ java {
     targetCompatibility = JavaVersion.VERSION_1_8
 }
 
+    project.plugins.withType(JavaPlugin::class) {
```
Comment (thc202) [inline build.gradle.kts:27]: This can be removed and just have the dependencies configuration.

File: build.gradle.kts [hunk 2]
```diff
@@ -22,6 +24,16 @@ java {
...
+        dependencies {
+            "errorprone"("com.google.errorprone:error_prone_core:2.7.1")
+            // When building ZAP releases it's also used Java 8.
+            if (JavaVersion.current() == JavaVersion.VERSION_1_8 || System.getenv("ZAP_RELEASE") != null) {
```
Comment (thc202) [inline build.gradle.kts:31]: The env var check is not be needed (unless you think in the near future the add-on is going to be included by default in ZAP packages).

File: build.gradle.kts [hunk 3]
```diff
@@ -30,7 +42,16 @@ spotless {
...
+tasks.withType<JavaCompile>().configureEach { 
+    options.encoding = "utf-8"
+    options.errorprone {
+            disableAllChecks.set(true)
+            error(
+                "MissingOverride",
+                "WildcardImport"
+            )
+        }
+    }
```
Comment (thc202) [inline build.gradle.kts:54]: nit: could be shifted left. (You could configure Spotless to check the build file too.)

**Comment by kingthorin** [comment]
@thc202 better?

**Comment by thc202** [comment]
LGTM

**Comment by kingthorin** [comment]
Thanks

**Comment by thc202** [comment]
No problem.

**Pull request merged**